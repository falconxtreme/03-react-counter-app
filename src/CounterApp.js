import React, { useState } from "react";
import PropTypes from "prop-types";

//Functional Component
const CounterApp = ({ value }) => {
  // const state = useState('Goku');
  // console.log(state);
  // const [nombre, setNombre] = useState("Goku");
  // console.log(nombre, setNombre);
  const [counter, setCounter] = useState(value);

  const handleAdd = (e) => {
    // setCounter(counter + 1);
    setCounter((c) => c + 1);
  };

  const handleReset = (e) => {
    setCounter(value);
  };

  const handleSubstract = (e) => {
    setCounter(counter - 1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubstract}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 10,
};

export default CounterApp;
