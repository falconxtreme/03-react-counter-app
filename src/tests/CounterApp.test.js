import { shallow } from "enzyme";

import CounterApp from "../CounterApp";

describe("Pruebas de <CounterApp />", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("debe de mostrar <ConterApp /> correctamente con sus valores por defecto", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el valor por defecto de 10", () => {
    const wrapper = shallow(<CounterApp value={100} />);

    const numeroH2 = parseInt(wrapper.find("h2").text());

    expect(numeroH2).toBe(100);
  });

  test("debe de incrementar con el boton +1", () => {
    // const btn1 = wrapper.find('button').at(0);
    wrapper.find("button").at(0).simulate("click");

    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("11");
  });

  test("debe de reseteat con el valor de 10", () => {
    // const btn1 = wrapper.find('button').at(0);
    wrapper.find("button").at(1).simulate("click");

    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("10");
  });

  test("debe de decrementar con el boton -1", () => {
    // const btn1 = wrapper.find('button').at(0);
    wrapper.find("button").at(2).simulate("click");

    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("9");
  });
});
