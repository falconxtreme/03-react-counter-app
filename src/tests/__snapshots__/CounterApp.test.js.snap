// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <CounterApp /> debe de mostrar <ConterApp /> correctamente con sus valores por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp!
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
